src/mrpro/operators/_FourierOp.py

```diff
@@ -58,38 +56,46 @@ def __init__(
 
         super().__init__()
 
-        def get_dims_for_traj_type(traj_type):
-            return [dim for dim in (-3, -2, -1) if traj.traj_type_along_kzyx[dim] == traj_type]
-
         def get_spatial_dims(spatial_dims, dims):
             return [s for s, i in zip((spatial_dims.z, spatial_dims.y, spatial_dims.x), (-3, -2, -1)) if i in dims]
 
         def get_traj(traj, dims):
             return [k for k, i in zip((traj.kz, traj.ky, traj.kx), (-3, -2, -1)) if i in dims]
 
-        # Find dimensions which do not require any transform
-        self._ignore_dims = get_dims_for_traj_type(TrajType.SINGLEVALUE)
+        self._ignore_dims, self._fft_dims, self._nufft_dims = [], [], []
+        for dim, type in zip((-3, -2, -1), traj.type_along_kzyx):
+            if type & TrajType.SINGLEVALUE:
+                # dimension which do not require any transform
+                self._ignore_dims.append(dim)
+            elif type & TrajType.ONGRID:
+                self._fft_dims.append(dim)
+            else:
+                self._nufft_dims.append(dim)
 
-        # Find dimensions which require FFT
-        self._fft_dims = get_dims_for_traj_type(TrajType.ONGRID)
-        if len(self._fft_dims) > 0:
+        if self._fft_dims:
             self._fast_fourier_op = FastFourierOp(
-                dim=self._fft_dims,
+                dim=tuple(self._fft_dims),
                 recon_shape=get_spatial_dims(recon_shape, self._fft_dims),
                 encoding_shape=get_spatial_dims(encoding_shape, self._fft_dims),
             )
 
         # Find dimensions which require NUFFT
-        self._nufft_dims = get_dims_for_traj_type(TrajType.NOTONGRID)
-        if len(self._nufft_dims) > 0:
-            # Special case when the fft dimension does not align with the corresponding k2, k1 or k0 dimension.
-            # E.g. kx is of shape (1,1,30,1,1): kx sampling along k2.
-            if self._fft_dims != [dim for dim in (-3, -2, -1) if traj.traj_type_along_k210[dim] == TrajType.ONGRID]:
+        if self._nufft_dims:
+            fft_dims_k210 = [
+                dim
+                for dim in (-3, -2, -1)
+                if (traj.type_along_k210[dim] & TrajType.ONGRID)
+                and not (traj.type_along_k210[dim] & TrajType.SINGLEVALUE)
+            ]
+            if self._fft_dims != fft_dims_k210:
```

Comment:
This checked here because as far as I can see, there is only a problem if the FFT dimensions are not aligned and a combination of FFT and NUFFT is carried out. Maybe we can rephrase the error message to make this clearer.